lib/rate_tracker.ex

```diff
@@ -107,6 +107,10 @@ defmodule Instruments.RateTracker do
       {_key, 0} -> false
       {_key, _rate} -> true
     end)
+    |> Enum.map(fn {key, count} ->
+      report_interval_seconds = @report_interval_ms / 1000
+      {key, count / report_interval_seconds}
```

Comment:
this isn't technically correct either, right? The last report may have been only 5 ms ago, in which case the rate should be the count divided by 5ms

You could have `dump_rates` go through the genserver and use state.last_update_time. It would be like doing dump_and_clear_counts but without clearing

Reply:
Yea, that's a good point. I guess there's no harm in it going through the GenServer